variables.tf

```diff
@@ -1,6 +1,11 @@
 variable "kms_key_arn" {
   description = "The ARN of the KMS key to use with the Lambda function"
   type        = string
+
+  validation {
+    condition     = can(regex("arn:aws:kms:[a-z0-9-]+:[0-9]+:key/[a-f0-9-]+", var.kms_key_arn))
```

Comment:
Does this validation works if the key does not exist yet? If its created together with this module? In this case only a reference is passed so i'm worried that this validation will fail in this case.

Reply:
hmm, you're right that on fresh deploys this id is only known after apply. As discussed: I will remove it for now. I can always look into improving this one later.